### PR TITLE
Cleanup some code and improve REAMDE

### PR DESCRIPTION
**Pull request by mathbunnyru**
Supersmall changes.
I decided to read all the code here and cleaned it up a little bit.
Also, cleaned up the code in README and fixed broken (and redirect) links.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,14 +15,15 @@ jobs:
   # https://github.com/pre-commit/action
   pre-commit:
     name: Lint
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```
Comment (mathbunnyru) [inline .github/workflows/test.yml:18]: Makes it consistent with the next job.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -15,14 +15,15 @@ jobs:
...
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
       - uses: pre-commit/action@v3.0.0
 
   # Run tests
   test:
+    name: Test
```
Comment (mathbunnyru) [inline .github/workflows/test.yml:26]: Looks better on GitHub.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -47,11 +48,9 @@ jobs:
         run: cat ci/config_token.py >> ${{ matrix.example }}/jupyterhub_config.py
 
       - name: Run JupyterHub
-        run: |
-          cd ${{ matrix.example }}
-          docker-compose up -d
+        working-directory: ${{ matrix.example }}
```
Comment (mathbunnyru) [inline .github/workflows/test.yml:51]: A bit cleaner logs.

File: basic-example/Dockerfile.jupyterhub [hunk 4]
```diff
@@ -4,9 +4,9 @@ ARG JUPYTERHUB_VERSION
 FROM jupyterhub/jupyterhub:$JUPYTERHUB_VERSION
 
 # Install dockerspawner, nativeauthenticator
+# hadolint ignore=DL3013
 RUN python3 -m pip install --no-cache-dir \
-        dockerspawner==12.* \
-        jupyterhub-nativeauthenticator==1.*
+    dockerspawner \
+    jupyterhub-nativeauthenticator
```
Comment (mathbunnyru) [inline basic-example/Dockerfile.jupyterhub:10]: We're testing latest jupyterhub version on schedule, so it should work with latest spawner and authenticator.

File: basic-example/docker-compose.yml [hunk 5]
```diff
@@ -32,13 +32,11 @@ services:
       # All containers will join this network
       DOCKER_NETWORK_NAME: jupyterhub-network
       # JupyterHub will spawn this Notebook image for users
-      DOCKER_NOTEBOOK_IMAGE: jupyter/minimal-notebook:latest
+      DOCKER_NOTEBOOK_IMAGE: jupyter/base-notebook:latest
```
Comment (mathbunnyru) [inline basic-example/docker-compose.yml:35]: `jupyter/base-notebook` is smaller and is a base image, so it's better candidate for testing.

File: basic-example/docker-compose.yml [hunk 6]
```diff
@@ -32,13 +32,11 @@ services:
...
       # Notebook directory inside user image
       DOCKER_NOTEBOOK_DIR: /home/jovyan/work
       # Using this run command
       DOCKER_SPAWN_CMD: start-singleuser.sh
-    command: >
-      jupyterhub -f /srv/jupyterhub/jupyterhub_config.py
```
Comment (mathbunnyru) [inline basic-example/docker-compose.yml:41]: This is part of the Docker image

File: basic-example/Dockerfile.jupyterhub [hunk 7]
```diff
@@ -4,9 +4,9 @@ ARG JUPYTERHUB_VERSION
...
 
-COPY ./jupyterhub_config.py /srv/jupyterhub/jupyterhub_config.py
```
Comment (mathbunnyru) [inline basic-example/Dockerfile.jupyterhub:11]: I was wrong here, this file is being mounted as part of docker-compose as well.
So, copy here makes no sense.

**Review by manics — APPROVED** [review]
Thanks!

**Pull request merged**